src/main/java/fi/aalto/cs/apluscourses/intellij/model/CourseUpdater.java

```diff
@@ -116,7 +116,16 @@ protected void doTask() {
       return;
     }
     progress.increment();
-    updateModules(fetchModulesInfo());
+    JSONObject newCourseConfig = null;
+    try {
+      newCourseConfig = fetchCourseConfiguration();
+    } catch (JSONException | IOException e) {
+      progress.finish();
+      return;
+    }
+    progress.increment();
```

Comment:
Nit: this logic could perhaps somehow be moved to another function.

Reply:
That hypothetical function would either need to have these two exceptions as a checked exception (which will result in the same code as currently) or catch exceptions and return null (which would require a null check instead of a catch-clause; all in all, won't make things simpler)